ide/gsf.testrunner/src/org/netbeans/modules/gsf/testrunner/TestCreatorProviderProcessor.java

```diff
@@ -47,11 +47,11 @@ protected boolean handleProcess(Set<? extends TypeElement> annotations, RoundEnv
             if(registration == null) {
                 continue;
             }
-            File f = layer(e).instanceFile("Services", null);
+            File f = layer(e).instanceFile("Services/TestCreatorProviders", null);
             f.stringvalue("instanceOf", TestCreatorProvider.class.getName());
             f.bundlevalue("displayName", registration.displayName());
             f.bundlevalue("identifier", registration.identifier());
-            if (registration.position() != -1) {
+            if (registration.position() != -1 && registration.position() != Integer.MAX_VALUE) {
```

Comment:
The Processor generates same position for all position-less registrations, which fails commit-validationtest.